build-cmd.sh

```diff
@@ -50,31 +47,102 @@ pushd "$top/luau"
     case "$AUTOBUILD_PLATFORM" in
         windows*)
             set -o igncr
-            opts="$LL_BUILD_RELEASE /EHsc"
-            cmake -G "$AUTOBUILD_WIN_CMAKE_GEN" -A "$AUTOBUILD_WIN_VSPLATFORM" \
+            opts="$(replace_switch /Zi /Z7 $LL_BUILD_RELEASE) /EHsc"
+            plainopts="$(remove_switch /GR $(remove_cxxstd $opts))"
+
+            cmake -G "Ninja" -DCMAKE_BUILD_TYPE="Release" \
                   -DCMAKE_INSTALL_PREFIX="$(cygpath -m "$stage")" \
                   -DCMAKE_C_FLAGS="$(remove_cxxstd $opts)" \
                   -DCMAKE_CXX_FLAGS="$opts" \
                   ../luau
-            cmake --build . -- /p:Configuration=Release
-            cmake --build . --target Luau.Repl.CLI -- /p:Configuration=Release
+            cmake --build . --config Release
+
+            # conditionally run unit tests
+            if [ "${DISABLE_UNIT_TESTS:-0}" = "0" ]; then
+                ./Luau.UnitTest.exe
+                ./Luau.Conformance.exe
+                ./Luau.UnitTest.exe --fflags=true
+                ./Luau.Conformance.exe --fflags=true
+                ./Luau.Conformance.exe -O2
+                ./Luau.Conformance.exe -O2 --fflags=true
+                ./Luau.Conformance.exe --codegen
+                ./Luau.Conformance.exe --codegen --fflags=true
+                ./Luau.Conformance.exe --codegen -O2
+                ./Luau.Conformance.exe --codegen -O2 --fflags=true
+                ./luau ../luau/tests/conformance/assert.lua
+                ./luau-analyze ../luau/tests/conformance/assert.lua
+                ./luau-compile ../luau/tests/conformance/assert.lua
+            fi
 
             mkdir -p "$stage/bin"
 
-            cp -v "Release/Luau.Ast.lib" "$stage/lib/release/"
-            cp -v "Release/Luau.CodeGen.lib" "$stage/lib/release/"
-            cp -v "Release/Luau.Compiler.lib" "$stage/lib/release/"
-            cp -v "Release/Luau.Config.lib" "$stage/lib/release/"
-            cp -v "Release/Luau.VM.lib" "$stage/lib/release/"
+            cp -v "Luau.Ast.lib" "$stage/lib/release/"
+            cp -v "Luau.CodeGen.lib" "$stage/lib/release/"
+            cp -v "Luau.Compiler.lib" "$stage/lib/release/"
+            cp -v "Luau.Config.lib" "$stage/lib/release/"
+            cp -v "Luau.VM.lib" "$stage/lib/release/"
+
+            cp -v luau.exe "$stage/bin/"
+        ;;
+        darwin*)
```

Comment:
So much of this is replicated between the three platform stanzas, it seems as though after the specific build commands, the rest could be moved to the common code following `esac`.

Of course you'd need platform-specific
```
exe='.exe'
function lib {
    echo "$1.lib"
}
```
vs.
```
exe=''
function lib {
    echo "lib$1.a"
}
```
to unify the differences in the replicated commands.

While we're at it, though, why build `luau.exe` only on Windows? I've found it useful to be able to run the standalone REPL on Mac too.